glitch/parsers/cmof.py

```diff
@@ -691,41 +727,86 @@ def __init__(self, source, ast):
+                            assoc.args[0], self.source),
+                            assoc.args[1])
                 else:
                     value = ChefParser._get_content(value_ast, self.source)
                     has_variable = ChefParser._check_has_variable(value_ast)
-                    if value == "nil": 
+                    if value == "nil":
                         value = ""
                         has_variable = False
-                    variable = Variable(current_name, value, has_variable)
-                    variable.line = ChefParser._get_content_bounds(key, self.source)[0]
-                    variable.code = ChefParser._get_source(ast, self.source)
-                    self.variables.append(variable)
+
+                    variable = create_variable(
+                        key, current_name, value, has_variable)
+
+                    if parent == None:
+                        self.variables.append(variable)
+                    else:
+                        parent.variables.append(variable)
 
             if ChefParser._check_node(ast, ["assign"], 2):
+                def get_parent(parent_name, vars):
+                    for var in vars:
+                        if var.name == parent_name:
+                            return var
+                    return None
+
                 name = ""
-                names = ChefParser._get_content(ast.args[0], self.source).split("[")
-                for n in names:
+                names = ChefParser._get_content(
+                    ast.args[0], self.source).split("[")
+                parent = None
+                for i, n in enumerate(names):
                     if n.endswith("]"):
                         n = n[:-1]
                     if (n.startswith("'") and n.endswith("'")) or \
                             (n.startswith('"') and n.endswith('"')):
-                        name += n[1:-1]
+                        name = n[1:-1]
                     elif n.startswith(":"):
-                        name += n[1:]
+                        name = n[1:]
                     else:
-                        name += n
+                        name = n
 
-                    name += "."
-                name = name[:-1]
-                parse_variable(ast, ast.args[0], name, ast.args[1])
+                    if i == len(names) - 1:
+                        parse_variable(
+                            parent, ast, ast.args[0], name, ast.args[1])
+                    else:
+                        if i == 0:
+                            parent = get_parent(name, self.variables)
+                            if parent == None:
+                                variable = create_variable(
+                                    ast.args[0], name, None, False)
+                                self.variables.append(variable)
+                                parent = variable
+                            else:
+                                pass
```

Comment:
When does this happen?

Reply:
It was supposed to happen while parsing the second line in this example:
`default[:social_graph][:deploy_path]` = "/data/apps/social_graph"
default[:social_graph][:version] = "1.0.8"`
But now, when you mentioned it, i figure it out that for some reason that i do not understand, self.variables is empty after parsing the first line, instead of containing a variable named default, composed by a variable named social_graph, composed by a variable named deploy_path with value "/data/apps/social_graph".